### PR TITLE
Test improvements: MSW, Old vs New Multicaller and fetch error traces 

### PR DESCRIPTION
**Pull request by agualis**
# Description

Includes new tests and several refactors to make the code more testable and debuggable:

- New `handleFetchException` helper to use console.trace instead of console.error to get the real line where the exception is originated: 
console trace
- New eslint rule to enforce using DI for "Old Multicaller" (we will soon do the same with the "new Multicaller")
- Separation of REST and Graphql MSW handlers 
- Added `name` parameter to add names to Grapql queries so that they can be intercepted by MSW 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/services/balancer/subgraph/entities/poolShares/index.ts [hunk 1]
```diff
@@ -14,7 +14,8 @@ export default class PoolShares {
   }
 
   public async get(args = {}, attrs = {}): Promise<PoolShare[]> {
-    const query = this.query(args, attrs);
+    const queryName = 'PoolShares';
+    const query = this.query(args, attrs, queryName);
     const data = await this.service.client.get(query);
```
Comment (agualis) [inline src/services/balancer/subgraph/entities/poolShares/index.ts:19]: In this type of manual graphql clients` client.get` returns `Promise<any>` and receives `Record<string, any>`, which makes it unsafe and less testable. 

I know that we have some complex queries where we do complex stuff like decorating but, I think that we could improve other queries like this ones by using a TS typed Graphql client like shown in this spike: 
https://github.com/balancer-labs/frontend-v2/pull/2380

 @garethfuller Do we talk about this?

Comment (agualis) [reply src/services/balancer/subgraph/entities/poolShares/index.ts:19]: Also, we have similar logic in balancer-subgraph.client.ts, gauges-subgraph.client.ts, block-subgraph.client.ts.
Duplication is not always bad but in this case they were having different behaviour: 

One was sending query as an object and another one was sending the JSON.stringified object. 

Using a single graphql client could avoid those problems.

File: src/lib/utils/exceptions.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export function handleFetchException(
```
Comment (garethfuller) [inline src/lib/utils/exceptions.ts:1]: Should maybe be more explicit with the function name here, e.g.
```suggestion
export function logException(
```

File: src/composables/queries/useUserGaugeSharesQuery.spec.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import { mountComposable, waitForQueryData } from '@tests/mount-helpers';
+import {
+  defaultGaugeBalance,
+  defaultPoolId,
+  defaultTotalSupply,
+} from '@tests/msw/graphql-handlers';
+
+initDependenciesWithDefaultMocks();
+
+test('Returns token prices from balancer SDK', async () => {
```
Comment (garethfuller) [inline src/composables/queries/useUserGaugeSharesQuery.spec.ts:12]: Should this description be changed?

File: src/composables/queries/useUserPoolSharesQuery.spec.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import useUserPoolSharesQuery from '@/composables/queries/useUserPoolSharesQuery';
+import { mountComposable, waitForQueryData } from '@tests/mount-helpers';
+import { defaultPoolBalance } from '@tests/msw/graphql-handlers';
+
+test('Returns token prices from balancer SDK', async () => {
```
Comment (garethfuller) [inline src/composables/queries/useUserPoolSharesQuery.spec.ts:5]: And this one?

File: src/services/balancer/gauges/gauges-subgraph.client.ts [hunk 5]
```diff
@@ -21,7 +21,7 @@ export class GaugesSubgraphClient {
   }
 
   public payloadFor(query) {
-    return JSON.stringify({ query: jsonToGraphQLQuery({ query }) });
```
Comment (garethfuller) [inline src/services/balancer/gauges/gauges-subgraph.client.ts:24]: I assume this is just redundant? If not will it not break the request?

File: src/services/balancer/subgraph/entities/poolShares/index.ts (hunk 1 above)
Comment (garethfuller) [reply src/services/balancer/subgraph/entities/poolShares/index.ts:19]: Yeah, this pattern has grown arms and legs from the initial implementation. We should use a generator like in your spike, but will require quite a big refactor.

**Review by garethfuller — APPROVED** [review]

File: src/lib/utils/exceptions.ts (hunk 2 above)
Comment (agualis) [reply src/lib/utils/exceptions.ts:1]: I wanted to explicitly use this helper for fetch exceptions so what about `logFetchException`?

The current implementation can be valid for any other kind of exception but that could change if we decide to give more fetch related info in the future. Maybe a bit YAGNI but not too much effort.

File: src/services/balancer/gauges/gauges-subgraph.client.ts (hunk 5 above)
Comment (agualis) [reply src/services/balancer/gauges/gauges-subgraph.client.ts:24]: I think so, also it was preventing MSW from detecting this requests as Graphql ones.

**Pull request merged**